### ISSUE TITLE
Overfitting during training on MegaDepth

### ISSUE THREAD

**Issue by cy69855522** [magicleap/SuperGluePretrainedNetwork#81]
Hi, thanks for the wonderful works. Recently, I tried to train superglue directly on megadepth and verify it on yfcc, but there was a serious over-fitting phenomenon. I keep the number of matching pairs consistent with the number of unmatched pairs during training, and find that the effect has improved, but it is still far from the results reported in the paper. I wonder if you adopted a strategy like dropout in training?

**Comment by Skydes** [comment]
A little bit of overfitting is expected but the task is sufficiently low-level so that the model generalizes well. We did not apply dropout or other forms of regularization. We however added some random points doing training. We did so for batching reasons but this could also have a regularization effect, although we did not investigate this in detail and I don't expect this to make much difference.

**Comment by uekoh** [comment]
Hi!
I have a question.

**added some random points** 
Is this mean 
**added some random ""unmatching"" points** ?

**Comment by cy69855522** [comment]
> Some overfitting is expected due to the large size of the model but the task is sufficiently low-level so that the model generalizes well. We did not apply dropout or other forms of regularization. We however added some random points doing training. We did so for batching reasons but this could also have a regularization effect, although we did not investigate this in detail and I don't expect this to make much difference.

Thanks for the reply. But I encountered a serious overfitting problem. Could you please tell me how many pairs have you taken for each scene on MegaDepth? （For ScanNet, it is 200）.

**Comment by cy69855522** [comment]
> Hi!
> I have a question.
> 
> **added some random points**
> Is this mean
> **added some random ""unmatching"" points** ?

points with 0 confidence

**Comment by cy69855522** [comment]
> Some overfitting is expected due to the large size of the model but the task is sufficiently low-level so that the model generalizes well. We did not apply dropout or other forms of regularization. We however added some random points doing training. We did so for batching reasons but this could also have a regularization effect, although we did not investigate this in detail and I don't expect this to make much difference.

I have finished the work, thanks very much.
